client/src/document/index.scss

```diff
@@ -38,16 +38,24 @@
     a:active {
       background-color: transparent;
     }
-
-    a[href^="#"]:hover {
-      &:after {
+    
+    a[href^="#"] {
+      &::before {
         color: var(--text-inactive);
         content: "#";
         display: inline-flex;
```

Comment:
Not an issue with your code, but something I noticed testing on Chromium which it would be nice to fix now: `display: inline-flex` seems to underline this with its parent text-decoration, whereas setting this to `display: inline-block` clears this:
```suggestion
        display: inline-block;
```
before:
![image](https://user-images.githubusercontent.com/755354/229100571-ffc648e0-6af2-4088-adf7-d1e578b74b16.png)

after:
![image](https://user-images.githubusercontent.com/755354/229100483-ba93ba68-eff2-45ec-839b-2ce7ae499d11.png)